src/util/file_download.ts

```diff
@@ -0,0 +1,27 @@
+import { isAndroid } from "./is_android";
+import { isSafari } from "./is_safari";
+
+export const fileDownload = (
+  element: HTMLElement,
+  href: string,
+  filename: string
+): void => {
+  const a = document.createElement("a");
+  a.target = "_blank";
+  a.href = href;
+  a.download = filename;
+
+  element.shadowRoot!.appendChild(a);
+
+  a.addEventListener("click", () => {
+    element.shadowRoot!.removeChild(a);
+  });
```

Comment:
Because of this event listener `element` and `a` can never be garbage collected, meaning we have a memory leak. We should remove the event listener.